com.basistech.m2e.code.quality.shared/src/main/java/com/basistech/m2e/code/quality/shared/AbstractMavenPluginConfigurationTranslator.java

```diff
@@ -140,6 +143,22 @@ protected URL resolveLocation(String resc) {
 		return this.resourceResolver.resolveLocation(resc);
 	}
 
+	protected void copyIfChanged(InputStream input, Path output) throws IOException {
```

Comment:
I'm not very familiar with the `org.codehaus.plexus.util.IOUtil` that is used here, but it strikes me as a little curious how we have a `File` (or `Path`) here and create an `InputStream` and a `byte[]` to be able to use `IOUtil.contentEquals` and `Files.copy` - I don't suppose there is a more direct way? Just asking, also fine to merge it as is.

PS: Also surprised this doesn't use the Eclipse APIs (IPath and IFile and what not), but that's already the case as-is, so certainly out of scope for this change! :smile:

Reply:
I tried to use NIO api wherever possible (BTW: `Files.copy` is NIO, too). As `IOUtils` is available, I'm using it here as there is no such thing in NIO. I'm open to better suggestions, if there are any.

I thought about Eclipse API too (it already has something like that built-in), but that would be a larger refactoring...

Reply:
OK, sure; you can always raise more PRs for the larger refactoring... let me get this in as is, for now.